src/chainstate/burn/operations/mod.rs

```diff
@@ -150,3 +260,26 @@ where
     }
 }
 
+
+// parser helpers
+pub fn parse_u32_from_be(bytes: &[u8]) -> Option<u32> {
```

Comment:
The rust u32 and u16 primitives provide big endian parse functions that should be used -- not just because we shouldn't reimplement primitives, but also because it won't require casting the u8s to u32s for bitshifts

```rust
fn parse_u32_from_be(bytes: &[u8]) -> Option<u32> {
    x.try_into().ok().map(u32::from_be_bytes)
}
```

Reply:
Didn't know that at the time it was written.  Will fix.

Reply:
Fixed in 8f8c6e10